pkg/api/resource/quantity.go

```diff
@@ -502,7 +519,7 @@ func (q *Quantity) RoundUp(scale Scale) bool {
 // Add adds the provide y quantity to the current value. If the current value is zero,
 // the format of the quantity will be updated to the format of y.
 func (q *Quantity) Add(y Quantity) {
-	q.s = nil
+	q.s = ""
```

Comment:
Added a test, also added one more guard for non-canonical strings (a
leading 0 in the shifted numerator+denominator cannot be canonical).
The parse routine always returns 0 for numerator (even when it has a
leading .) and so the left most position being 0 means the value
cannot be canonical.

On Tue, May 24, 2016 at 6:48 PM, Clayton Coleman <ccoleman@redhat.com> wrote:
> Yes.  We may need a more holistic quantity stress tester (I did one
> for add and remove, but we might want to get more formal).